security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.ts.openshift.security.basic;
+
+import jakarta.inject.Singleton;
+
+import io.quarkus.vertx.web.RouteFilter;
+import io.vertx.ext.web.RoutingContext;
+
+@Singleton
+public class SecurityFilter {
+
+    @RouteFilter(401)
+    void addAuthHeader(RoutingContext routingContext) {
+        try {
+            if (routingContext.request().headers().get("Sec-WebSocket-Protocol") != null) {
```

Comment:
What is a purpose of this? I don't understand it. 
If I understood the issue https://github.com/quarkusio/quarkus/issues/40307 correctly, then this is just a workaround for issue that is fixed in 3.10.1. Is that correct?
If yes, do we really want to have a workaround in our TS? At least, it should have a comment linking the issue, fix and explaining it.
WDYT @michalvavrik ?

Reply:
it never had anything to do with websockets. https://github.com/quarkusio/quarkus/issues/40307 is about existence of `@RouteFilter(404)`. You can have it empty and it will still be a reproducer. What you want to do is to have Reactive Route filter as then, Reactive Routes activates CDI request context.

Reply:
As Michal noted correctly, this is not a workaround, but condition for an issue to be reproduced. The condition, on the other hand is a workaround for Websockets, but that is another story :)

I will try to clean the filter from everything that is not required for the bug to resurface, though

Reply:
Everything related to websockets was removed from the filter